scripts/west_commands/runners/spsdk.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing with SPSDK.'''
+
+import logging
+import os
+import subprocess
+from pathlib import Path
+
+from runners.core import RunnerCaps, ZephyrBinaryRunner
+
+
+class SPSDKBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for SPSDK.'''
+
+    def __init__(
+        self, cfg, bootdevice=None, family=None, bootloader=None, flashbin=None, commander=None
+    ):
+        super().__init__(cfg)
+        self.file = cfg.file
+        self.file_type = cfg.file_type
+        self.hex_name = cfg.hex_file
+        self.bin_name = cfg.bin_file
+        self.elf_name = cfg.elf_file
+
+        self.bootdevice = bootdevice
+        self.family = family
+        self.bootloader = bootloader
+        self.flashbin = flashbin
+        self.commander = commander
+
+    @classmethod
+    def name(cls):
+        return 'spsdk'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--bootdevice', default='spl', help='boot device')
+        parser.add_argument('--family', required=True, help='family')
+        parser.add_argument('--bootloader', required=True, help='bootloader')
+        parser.add_argument('--flashbin', required=True, help='nxp container image flash.bin')
+        parser.add_argument(
+            '--commander',
+            default='nxpuuu',
+            help="SPSDK Commander, default is nxpuuu",
+        )
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return SPSDKBinaryRunner(
+            cfg,
+            bootdevice=args.bootdevice,
+            family=args.family,
+            bootloader=args.bootloader,
+            flashbin=args.flashbin,
+            commander=args.commander,
+        )
+
+    def do_run(self, command, **kwargs):
+        self.commander = os.fspath(Path(self.require(self.commander)).resolve())
+
+        if command == 'flash':
+            self.flash(**kwargs)
+
+    def flash(self, **kwargs):
+        if self.bootdevice == 'spl':
+            cmd = [self.commander] + ['run'] + [f"SDPS[-t 10000]: boot -f {self.flashbin}"]
```

Comment:
Actually it maybe use a uuu script to handle multiple containers, so is it possible to use a specific script according to different platform requirement?

Reply:
In my opinion, west flash is to program a bootable firmware to board.
What do you mean by multiple containers? You want to flash multiple images by west flash? Or any example of the script to share
If the script could be as an argument to use, I think you can send patch to extend the support when you have specific platofrm/case to use.
Thanks.

Reply:
Multiple containers will be used in order to boot U-Boot and Linux, NP, let's cover this feature in future.